src/url-params.ts

```diff
@@ -49,7 +49,7 @@ export class AnalyticsUrlParams {
 	 * @param queryString URL query string to process
 	 * @return a modified query string that hides processed/consumed parameters if the query string should be modified
 	 */
-	consumeUrlParameters(queryString: string): string | null {
+	consumeUrlParameters(queryString: string): string[] | null {
```

Comment:
Why do we need to pass a list here? 
To be clear, `consumeUrlParameters` uses the `setDeviceIds` function to parse and split string of `deviceId` values ([the relevant line](https://github.com/balena-io-modules/analytics-client/blob/3db19c67a74b61b7de33c6ddf2aa911d696a46c9/src/url-params.ts#L29))

Reply:
We dont, it stems from my confusion about setDeviceIds returning  the whole array instead of the first element of it.

Reply:
@gelbal @pranasziaukas I took a look again as to why I changed it to `string[]` and it was not because of expecting to recieve a list of device_ids from `setDeviceIds()` but because I want to return an array of strings that includes both the `passedDeviceId` and the `params.toString()` . The first one used for setting the client properly, the second one to build the clean URL with no `d_id` params.

Then again, I dont know if theres a better way on Typescript to handle this scenario

Reply:
I'd suggest using other functions to get these or implement new ones if necessary. So `AnalyticsUrlParams.allDeviceIds()` returns all the device identifiers as an array. The order is maintained, so the first identifier in the URL param is the first one in the array as well. The new generated identifiers are appended to the end of array.

Reply:
@gelbal @pranasziaukas  I was looking into change how the UI init works by using `AnalyticsUrlParams.allDeviceIds()` so as to keep `consumeUrlParams()` return value as before. But I stumble with the following issue:

Many of the methods like `allDeviceIds()` require that the `client` has already been passed to `AnalyticsUrlParams`. But since we are trying to init the `client`, we cant use this methods since there is no `client` to pass to them yet.

So basically my changes to `consumeUrlParams()` are not very good since Im just hardcoding it to bypass the need of a `client` but all the other methods on the class needs it.

What do you propose to do here?

Reply:
@laziob I don't think we have an issue here. Here is the `client` reference in [`allDeviceIds()`](https://github.com/balena-io-modules/analytics-client/blob/3db19c67a74b61b7de33c6ddf2aa911d696a46c9/src/url-params.ts#L91):
```js
const currentId = this.client != null ? this.client.deviceId() : null;`

So it should be fine if there is no `client` set. As far as I see `AnalyticsUrlParams` is designed to work without a `client` set. If we find a breaking case, let's update it for sure.

I suggest addressing the feedback on the UI PR and update the code not to depend on `consumeUrlParams`'s output to get `deviceId`. Instead `allDeviceIds()[0]` should return either the first `d_id` in URL params or null. 

Let's test these together with the changes here (putting aside the changes in `url-params`). I believe initializing the `client` with `deviceId` is a sufficient solution. Then once we have `client` initialized, we'll set it in `AnalyticsUrlParams`. I will test this solution on another component.

Reply:
@gelbal you were right, I made some changes on the UI repo to handle the `passedDeviceId` with `allDeviceIds()[0]` and it seems to work, both when a device id is passed and when its not. I already made the changes on a new commit on the UI PR. Please let me know what you think and if its okey I can revert the changes to `url-params-handler` here on a new commit.

Reply:
Nice. Looks great Ezequiel. Yes, please revert the changes in `url-params` here so we could ship this PR as well. Consecutively, we are not breaking backwards compatibility so we could roll the new version out at our own pace.

Reply:
Just to recap and make sure we're all on the same page, let's:
1. Initialize `AnalyticsUrlParams` without a client.
1. Use `AnalyticsUrlParams.consumeUrlParams` as needed.
1. Use `AnalyticsUrlParams.allDeviceIds` to get the relevant `deviceId`.
1. Initialize the `Client` with the `deviceId`.
1. Inject the client using `AnalyticsUrlParams.setClient`.

Voila!

Reply:
@pranasziaukas  yes I think thats about right, although I havent used de `AnalyticsUrlParams.setClient` on the UI changes.  I think it should run as `urlParamsHandler.setClient(analyticsClient)` right above [this line](https://github.com/balena-io/balena-ui/blob/9804d2216e7af2ebb4c4acf7abc4255f5a7e3efd/src/services/eventLog.ts#L36) but then in order to use it later on, the `initAnalyticsClient()` should return the `urlParamsHandler` instead of the `analyticsClient` or both?

Reply:
> please revert the changes in url-params here so we could ship this PR as well

@gelbal done!

Reply:
I like that summary @pranasziaukas noted here. @laziob do you mind updating the docs (Readme) to reflect this? 

So we explain the current practice outside of this PR as well.

Reply:
Sure, I'll update it asap.